### PR TITLE
feat(auth-server): remove sources on payment update

### LINKED ISSUE #2348: Extract 2FA QR code screen into a mixin
One step along the way to enabling 2FA as part of the login flow. Refs #2024

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-308)
┆Issue Number: FXA-308

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* When updating a customer with a payment method, we don't want any
  sources from the Sources API around.

This commit:

* Removes all sources from a customer after updating the default
  payment method with a payment method id.

Closes #2348

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 1]
```diff
@@ -299,6 +299,34 @@ export class StripeHelper {
...
+    if (sources.data.length === 0) {
+      return [];
+    }
+    return Promise.all(
+      sources.data.map(
+        (s) =>
+          (this.stripe.customers.deleteSource(
+            customerId,
+            s.id
+          ) as unknown) as Stripe.Card
```
Comment (chenba) [inline packages/fxa-auth-server/lib/payments/stripe.ts:325]: I think this should be `as Promise<Stripe.Card>` as the type of `this.stripe.customers.deleteSource` is `Promise<T>` (where `T` is a sum type of types in `Stripe`, one of which is `Stripe.Card`).

File: packages/fxa-auth-server/test/local/payments/stripe.js [hunk 2]
```diff
@@ -243,6 +243,32 @@ describe('StripeHelper', () => {
...
+    it('removes all the sources', async () => {
+      const ids = {
+        data: [{ id: uuidv4() }, { id: uuidv4() }, { id: uuidv4() }],
+      };
+      sandbox.stub(stripeHelper.stripe.customers, 'listSources').resolves(ids);
+      sandbox.stub(stripeHelper.stripe.customers, 'deleteSource').resolves({});
+      const result = await stripeHelper.removeSources('cust_new');
+      assert.deepEqual(result, [{}, {}, {}]);
+      sinon.assert.calledThrice(stripeHelper.stripe.customers.deleteSource);
+    });
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/payments/stripe.js:256]: nit-ish:
 - the test above doesn't check to see if the ids passed into `deleteSource` are the ones from the test.
 - the empty sources list branch of the code in `removeSources` isn't tested.

File: packages/fxa-auth-server/test/local/routes/subscriptions.js [hunk 3]
```diff
@@ -936,10 +936,16 @@ describe('DirectStripeRoutes', () => {
       directStripeRoutesInstance.stripeHelper.customer
         .onCall(1)
         .resolves(expected);
-
+      directStripeRoutesInstance.stripeHelper.customer.resolves({ data: [] });
```
Comment (chenba) [inline packages/fxa-auth-server/test/local/routes/subscriptions.js:939]: Why is this necessary?  I don't see an additional call to `stripeHelper.customer` above.

File: packages/fxa-auth-server/lib/payments/stripe.ts (hunk 1 above)
Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:325]: Yup, updated.

**Comment by 6a68** [comment]
I'm thinking this was intended to close Jira bug FXA-2348, which means this PR closes #6041 on github :+1:
